src/jit/morph.cpp

```diff
@@ -12422,6 +12422,33 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 }
             }
 
+            //
+            // Here we look for the following transformation
+            //
+            //                        EQ/NE                 EQ/NE
```

Comment:
I think a normal expression representation would be better than fancy trees. They take a lot of space and sometimes break if code is somehow reformatted.